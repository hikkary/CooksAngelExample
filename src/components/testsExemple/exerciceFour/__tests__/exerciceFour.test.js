import React from 'react';
import ExerciceFour from '..';

import { shallow } from 'enzyme';

it('should match the snapshot (good Practice)', () => {
  const component = shallow(
    <ExerciceFour counterTitle="test" counterNumber={0} />
  );

  expect(component).toMatchSnapshot();
});

it('should have two buttons', () => {
  const component = shallow(
    <ExerciceFour counterTitle="test" counterNumber={0} />
  );  const buttonElement = component.find('button');

  expect(buttonElement.length).toEqual(2);
});

it('should return null', () => {
  const component = shallow(
    <ExerciceFour counterTitle="test" counterNumber={0} disabled={true} />
  );

  expect(component.type()).toBe(null);
});



it('should call decrement function when we click ', () => {
  const decrement = jest.fn();

  const component = shallow(
    <ExerciceFour counterTitle="test" counterNumber={0} decrement={decrement} />
  );
  component.find('button.decrementButton').simulate('click');

  expect(decrement).toHaveBeenCalled();
});

it('should call increment function when we click', () => {
  const increment = jest.fn();
  const component = shallow(
    <ExerciceFour counterTitle="test" counterNumber={0} increment={increment} />
  );
  component.find('button.incrementButton').simulate('click');

  expect(increment).toHaveBeenCalled();
});
