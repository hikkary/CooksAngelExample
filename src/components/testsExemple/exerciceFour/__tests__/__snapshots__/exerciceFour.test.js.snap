// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot (good Practice) 1`] = `
<div
  className="counterContainer"
>
  <div>
    <span
      className="commonText"
    >
      test
    </span>
  </div>
  <div
    className="counterNumberContainer"
  >
    <span
      className="commonText"
    >
      0
    </span>
  </div>
  <div>
    <button
      className="styledButton decrementButton"
    >
      <span
        className="commonText"
      >
        -
      </span>
    </button>
    <button
      className="styledButton incrementButton"
    >
      <span
        className="commonText"
      >
        +
      </span>
    </button>
  </div>
</div>
`;
